src/tests/win-app-driver/FancyZonesTests/EditorSettingsTests.cs

```diff
@@ -51,21 +56,160 @@ public void ZoneCount()
+
+                WindowsElement applyButton = WaitElementByAccessibilityId("ApplyTemplateButton");
+                applyButton.Click();
+                ShortWait();
+
+                Assert.AreEqual(editorShowSpacingValue, GetEditZonesSetting<bool>(editorShowSpacing));
+                Assert.AreEqual(editorSpacingValue, GetEditZonesSetting<string>(editorSpacing));
+            }
         }
 
-        private int getSavedZoneCount()
+        [TestMethod]
+        public void SpacingTestsInvalid()
+        {
+            ShortWait();
+            OpenFancyZonesSettings();
+
+            WindowsElement editorButton = WaitElementByXPath("//Button[@Name=\"Edit zones\"]");
+            editorButton.Click();
+
+            WindowsElement spaceAroundSetting = WaitElementByAccessibilityId("spaceAroundSetting");
+            bool editorShowSpacingValue = spaceAroundSetting.Selected;
+
+            string[] invalidValues = { "!", "/", "<", "?", "D", "Z", "]", "m", "}", "1.5", "2,5" };
+
+            string editorSpacingValue = GetEditZonesSetting<string>(editorSpacing);
+
+            foreach (string value in invalidValues)
+            {
+                editorButton.Click();
+
+                WindowsElement paddingValue = WaitElementByAccessibilityId("paddingValue");
+                ClearText(paddingValue);
+                paddingValue.SendKeys(value);
+
+                WindowsElement applyButton = WaitElementByAccessibilityId("ApplyTemplateButton");
+                applyButton.Click();
+                ShortWait();
+
+                Assert.AreEqual(editorShowSpacingValue, GetEditZonesSetting<bool>(editorShowSpacing));
+                Assert.AreEqual(editorSpacingValue, GetEditZonesSetting<string>(editorSpacing));
+            }
+        }
+
+        [TestMethod]
+        public void SpacingTestLargeValue()
+        {
+            string zoneSettings = File.ReadAllText(_zoneSettingsPath);
+
+            const string largeValue = "1000";
+
+            ShortWait();
+            OpenFancyZonesSettings();
+
+            WindowsElement editorButton = WaitElementByXPath("//Button[@Name=\"Edit zones\"]");
+            editorButton.Click();
+
+            WindowsElement paddingValue = WaitElementByAccessibilityId("paddingValue");
+            ClearText(paddingValue);
+            ClearText(paddingValue);
+            paddingValue.SendKeys(largeValue);
+
+            WindowsElement applyButton = WaitElementByAccessibilityId("ApplyTemplateButton");
+            applyButton.Click();
+
+            editorButton.Click();
+            WindowsElement editorWindow = null;
+            try
+            {
+                editorWindow = WaitElementByAccessibilityId("MainWindow1");
+            }
+            catch { }
+
+            bool result = editorWindow == null;
+
+            if (editorWindow != null)
+            {
+                editorWindow.SendKeys(Keys.Alt + Keys.F4);
+            }
+
+            ExitPowerToys();
```

Comment:
These things should better set in `TestCleanup`. If tests fails with exception, this string wouldn't be executed, and at `TestCleanup` it is guaranteed.